app/javascript/mastodon/features/ui/components/image_loader.js

```diff
@@ -1,6 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import Motion from '../util/optional_motion';
+import spring from 'react-motion/lib/spring';
+import PanoramaViewer, { getPanoramaData } from './panorama_viewer';
```

Comment:
Haven't looked at the size, but could this be lazy-loaded? Only a small fraction of images will be panoramas, so it does not really make sense to always load it (see https://github.com/tootsuite/mastodon/blob/master/app/javascript/mastodon/features/ui/util/async-components.js).

Reply:
The part that checks for panoramas can't be lazy loaded, but the WebGL renderer could. Not sure if it'd be worth it though, the webpack chunks here are at least 30kB and the uncompressed renderer is just 9kB

Reply:
oops I forgot about gl-matrix, which is huge. I can't find anything where the webpack chunks are defined though?

Reply:
They are defined [here](https://github.com/tootsuite/mastodon/blob/master/app/javascript/mastodon/features/ui/util/async-components.js) - for an example how to use see [the emoji picker](https://github.com/tootsuite/mastodon/blob/master/app/javascript/mastodon/features/compose/components/emoji_picker_dropdown.js#L4).